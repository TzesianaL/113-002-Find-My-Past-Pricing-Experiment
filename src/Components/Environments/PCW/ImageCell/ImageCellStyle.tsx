import styled from 'styled-components'

export const ImageCellStyle = styled.img`
  @media (max-width: 600px) {
    width: 33px;
    margin-top: 0em;
    margin-bottom: 0.3em;
  }
  width: 30px;
  margin-top: 0.5rem;
`

export const TextStyle = styled.div`
  @media (max-width: 600px) {
    max-width: max-content;
  }
  max-width: 100px;
  padding: 0px 4px 0px 4px;
  grid-column: 1 / span 2;
`
