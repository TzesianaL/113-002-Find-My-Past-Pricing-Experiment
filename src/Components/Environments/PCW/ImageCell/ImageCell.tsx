import React, { FC } from 'react'
import Cell, { CellProps } from '../Cell/Cell'
import { ImageCellStyle, TextStyle } from './ImageCellStyle'

export interface ImageCellProps extends Partial<CellProps> {
  text?: string
  imagePath?: string
}

const ImageCell: FC<ImageCellProps> = ({ text, imagePath, ...props }) => {
  if (imagePath === '') {
    throw new Error('Image is not provided')
  }
  return (
    <Cell {...props}>
      <ImageCellStyle src={imagePath} alt="" />
      <TextStyle>{text}</TextStyle>
    </Cell>
  )
}

export default ImageCell
