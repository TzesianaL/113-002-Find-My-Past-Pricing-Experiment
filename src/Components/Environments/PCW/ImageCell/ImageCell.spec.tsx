import React from 'react'
import 'jest-styled-components'
import '@testing-library/jest-dom/extend-expect'
import * as renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import logo from 'images/logos/lifepro-prime1.png'
import ImageCell from './ImageCell'

describe('<ImageCell/>', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<ImageCell imagePath={logo} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should display text only when it is provided', () => {
    render(<ImageCell imagePath={logo} text="test logo" />)

    expect(screen.getByText('test logo')).toBeInTheDocument()
  })

  it('should throw an error when image is not provided', () => {
    const mockError = jest.spyOn(console, 'error')
    mockError.mockImplementation(() => {})

    expect(() => render(<ImageCell imagePath="" />)).toThrowError(
      'Image is not provided'
    )

    mockError.mockRestore()
  })
})
