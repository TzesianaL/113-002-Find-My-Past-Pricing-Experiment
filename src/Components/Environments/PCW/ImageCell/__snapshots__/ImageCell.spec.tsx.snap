// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageCell/> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #000000;
  background: #ffffff;
  min-width: 100px;
  height: 90px;
  border: 1px solid;
  border-color: lightgray;
  word-wrap: break-word;
  font-weight: bolder;
  font-size: small;
}

.c1 {
  width: 30px;
  margin-top: 0.5rem;
}

.c2 {
  max-width: 100px;
  padding: 0px 4px 0px 4px;
  grid-column: 1 / span 2;
}

@media (max-width:600px) {
  .c0 {
    min-width: 180px;
  }
}

@media (max-width:600px) {
  .c1 {
    width: 33px;
    margin-top: 0em;
    margin-bottom: 0.3em;
  }
}

@media (max-width:600px) {
  .c2 {
    max-width: -webkit-max-content;
    max-width: -moz-max-content;
    max-width: max-content;
  }
}

<div
  className="c0"
>
  <img
    alt=""
    className="c1"
    src="lifepro-prime1.png"
  />
  <div
    className="c2"
  />
</div>
`;
